Gem/Code/Source/Weapons/WeaponGathers.cpp

```diff
@@ -0,0 +1,115 @@
+
+    IntersectFilter::IntersectFilter
+    (
+        const AZ::Transform& initialPose, 
+        const AZ::Vector3& sweep, 
+        HitStatic intersectStatic, 
+        HitDynamic intersectDynamic, 
+        HitMultiple intersectMultiple, 
+        const NetEntityIdSet& filteredNetEntityIds
+    )
+        : m_initialPose(initialPose)
+        , m_sweep(sweep)
+        , m_intersectStatic(intersectStatic)
+        , m_intersectDynamic(intersectDynamic)
+        , m_intersectMultiple(intersectMultiple)
+        , m_filteredNetEntityIds(filteredNetEntityIds)
+    {
+        Multiplayer::INetworkTime* networkTime = Multiplayer::GetNetworkTime();
+        if (networkTime->IsTimeRewound())
+        {
+            m_rewindFrameId = networkTime->GetHostFrameId();
+        }
+    }
+
+    bool GatherEntities
+    (
+        const GatherParams& gatherParams, 
+        const ActivateEvent& eventData, 
+        const NetEntityIdSet& filteredNetEntityIds, 
+        [[maybe_unused]] IntersectResults& outResults
+    )
+    {
+        const AZ::Transform& startTransform = eventData.m_initialTransform;
+        const AZ::Vector3    sweep = eventData.m_targetPosition - startTransform.GetTranslation();
+        const HitMultiple    hitMultiple = gatherParams.m_multiHit ? HitMultiple::Yes : HitMultiple::No;
+        [[maybe_unused]] const GatherShape&   intersectShape = gatherParams.m_gatherShape;
+
+        IntersectFilter filter(startTransform, sweep, HitStatic::Yes, HitDynamic::Yes, hitMultiple, filteredNetEntityIds);
+        //gNovaGame->GetNetworkPhysicalWorld().WorldIntersect(intersectShape, filter, outResultList);
+
+        return true;
+    }
+
+    ShotResult GatherEntitiesMultisegment
+    (
+        const GatherParams& gatherParams, 
+        const NetEntityIdSet& filteredNetEntityIds, 
+        float deltaTime, 
+        ActiveShot& inOutActiveShot, 
+        IntersectResults& outResults
+    )
+    {
+        // This only works when our cast is not instantaneous (it requires some positive, non-zero travel speed)
+        AZ_Assert(gatherParams.m_travelSpeed > 0.0f, "GatherEntitiesMultiSegment called with an invalid travel speed! This will fail, use the non-segmented gather path instead.");
+
+        ShotResult result = ShotResult::DoNotTerminate;
+
+        const AZ::Transform& startTransform = inOutActiveShot.m_initialTransform;
+        const AZ::Vector3 sweep = (inOutActiveShot.m_targetPosition - startTransform.GetTranslation()).GetNormalized();
+
+        // World gravity for our current location (making the currently safe assumption that it's constant over the duration of our trace)
+        //const AZ::Vector3& gravity = gatherParams.m_bulletDrop ? gNovaGame->GetNetworkPhysicalWorld().GetWorldGravity(a_InOutActiveShot.GetInitialTransform().m_Position) : AZ::Vector3::CreateZero();
```

Comment:
This looks like it can be removed

Reply:
same, need to query gravity for proper bullet drop simulation.. some games use tensor fields for gravity, so gravity can change throughout a level.. not sure if that's something we support but it's why we're querying for gravity against a particular point in the world